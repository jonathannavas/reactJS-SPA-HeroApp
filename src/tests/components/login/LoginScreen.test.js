import { mount } from "enzyme";
import { AuthContext } from "../../../auth/AuthContext";
import { LoginScreen } from "../../../components/login/LoginScreen";
import { types } from "../../../types/types";

describe('Tests on LoginScreen.js', () => {

    const history = {
        replace: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn()
    }

    const wrapper = mount( 
        <AuthContext.Provider value={contextValue}>
            <LoginScreen history={ history } /> 
        </AuthContext.Provider>
    );

    test('Should show correctly with snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should use dispatch and navigation', () => {
        const handleClick = wrapper.find('button').prop('onClick');
        handleClick();
        expect( contextValue.dispatch ).toHaveBeenCalledWith({
            type: types.login,
            payload: {
                name:'Jonathan'
            }
        });

        expect( history.replace ).toHaveBeenCalledWith('/');

        localStorage.setItem('lastPath', '/dc');
        handleClick();

        expect( history.replace ).toHaveBeenCalledWith('/dc');

    });



});