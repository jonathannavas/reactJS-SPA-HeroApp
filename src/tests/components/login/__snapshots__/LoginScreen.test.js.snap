// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on LoginScreen.js Should show correctly with snapshot 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Ingresar
  </button>
</div>
`;
